File: internal/domain/suspicious_activity.go
package domain

import "time"

type Severity string

const (
	LowActivity      Severity = "low"
	MediumActivity   Severity = "medium"
	HighActivity     Severity = "high"
	CriticalActivity Severity = "critical"
)

type SuspiciousActivity struct {
	ID           int64      `json:"id"`
	UserID       int64      `json:"user_id"`
	ActivityType string     `json:"activity_type"`
	IPAddress    string     `json:"ip_address"`
	UserAgent    string     `json:"user_agent"`
	Description  string     `json:"description"`
	Metadata     []byte     `json:"metadata"`
	Severity     string     `json:"severity"`
	Resolved     *bool      `json:"resolved"`
	CreatedAt    *time.Time `json:"created_at"`
}

type CreateSuspiciousActivityAction struct {
	UserID       int64
	ActivityType string
	IPAddress    string
	UserAgent    string
	Description  string
	Metadata     []byte
	Severity     *Severity
}

File: frontend/src/types/models/suspicious_activity.ts
export type SuspiciousActivity = {
	id: number
	user_id: number
	activity_type: string
	ip_address: string
	user_agent: string
	description: string
	metadata: Record<string, any> | null
	severity: 'low' | 'medium' | 'high' | 'critical'
	resolved: boolean | null
	created_at: string | null
}

export type ResolveSuspiciousActivityRequest = {
	activity_id: number
}
